android-kit-base/src/main/java/com/mparticle/kits/KitManagerImpl.java

```diff
@@ -1334,12 +1334,25 @@ public void execute(String viewName,
             try {
                 if (provider instanceof KitIntegration.RoktListener && !provider.isDisabled()) {
                     MParticleUser user = MParticle.getInstance().Identity().getCurrentUser();
-                    Map<String, Object> objectAttributes = new HashMap<>();
 
+                    JSONArray jsonArray = provider.getConfiguration().getPlacementAttributes();
```

Comment:
It seems to me that there's a lot of small steps going on in this function, but if something fails at any point during the execution, we'll just get a generic `Failed to call execute for kit` error message, which to me seems like it would be time consuming to investigate. Would it make sense to have some validation steps in between each operation and then throw a specific error message?

Such as if the `jsonArray` is empty, or if the `JSONObject obj = jsonArray.optJSONObject(i);` throws, we can have a specific `JSON parsing error`. Or if `attributes` is empty or invalid, we can have a specific message for that?